sstables/sstables.cc

```diff
@@ -1396,6 +1396,24 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+void sstable::maybe_discard_large_filter(uint64_t data_file_size) {
+    if (is_system_keyspace(_schema->ks_name())) {
+        // do not discard filters of system tables
+        return;
+    }
+    assert(has_component(component_type::Filter));
```

Comment:
Don't use `assert()`, most components of an sstable are optional, including the filter. This assert can bring a node down for no reason.
Just replace with an early return.

Reply:
Ack.